### ISSUE TITLE
Can not derive on structs which have reference to generic structs

### ISSUE THREAD

**Issue by orhanbalci** [colin-kiegel/rust-derive-builder#55]
Hello there;
I'm trying to derive builder for the following struct but it gives compiler error : 
```
/// Common properties of steering behaviors 
#[derive(Builder)]
pub struct SteeringBehavior<'a, T: 'a + Real> {
    /// ownew of this behavior upon which the calculations will occur
    pub owner: &'a Steerable<T>,
    /// is this behavior enabled
    pub enabled: bool,
    /// limitations on speed and velocity calculations
    pub limiter: &'a Limiter<T>,
}
```
Compiler error : 
```
error[E0309]: the parameter type `T` may not live long enough
--> src/steering_behavior.rs:6:10
   |
6 | #[derive(Builder)]
   | ^^^^^^^
   |
= help: consider adding an explicit lifetime bound `T: 'a`...
note: ...so that the reference type `&'a steerable::Steerable<T> + 'a` does not outlive the data it points at
```

**Comment by colin-kiegel** [comment]
Interesting - I tried it with the current master and I can reproduce it. 

Changing to a `where` clause helps
```rust
#[derive(Builder)]
pub struct SteeringBehavior<'a, T> where T: 'a + Real {
    // ...
}
```

- which version of `derive_builder` and Rust are you using?
- does the where clause work for you?

**Comment by orhanbalci** [comment]
-Versions I use : 
Latest Stable Rust : 1.16.0
derive_builder : 0.3.0
-where clause worked for me also

**Comment by colin-kiegel** [comment]
fixed with v0.4.2
